src/linear_algebra/affine_space.lean

```diff
@@ -400,6 +400,43 @@ end
 
 end affine_independent
 
+namespace affine_space
+
+variables (k : Type*) (V : Type*) (P : Type*) [ring k] [add_comm_group V] [module k V]
+variables [affine_space k V P]
+
+/-- A `simplex k V P n` is a collection of `n + 1` affinely
+independent points. -/
+structure simplex (n : ℕ) :=
```

Comment:
Have you already tested this definition in follow-up code? My fear is that you'll often run into trouble with simplices that are "obviously" the same, but not the same as structure. It might turn out to be easier to work with
```
def simplex {n} (points : fin (n+1) → P) := affine_independent k V points
```
Also... how are you planning to deal with edge cases when you start doing geometry? Very often it will happen that you perform some construction, and then afterwards you end up with a triangle, except that it might be degenerate. And then you case split, and the degenerate case is obvious for trivial reasons. But with these definitions, I fear that this will be harder, because the nondegeneracy is locked into the structure.

Reply:
Although I didn't at the time of the PR, I now have complete, `sorry`-free definitions of `circumcenter` and `circumradius` for any `simplex` in a `euclidean_affine_space`. I'm not sure equality of simplices is something that comes up often; the only obvious action there would be to add the obvious `ext` lemma that two simplices are equal if each pair of corresponding points is equal.

There are various cases without degeneracies where it seems useful for functions on a `simplex` to produce one with the same type and having the points as part of the type seems nastier (`simplex.medial`, where `simplex.medial.circumcenter` is the 3(n+1)-point center, `simplex.intouch`, etc.). As for cases such `triangle.pedal` where degeneracy is possible, you probably end up with two versions (one producing the indexed family of points, one producing a `simplex` but with extra hypotheses that need to be passed, such as the point passed as an argument not being on the circumcircle in the case of `triangle.pedal`, and in that case there would be the third variant `triangle.simson_line` with a hypothesis that the point passed as the argument is on the circumcircle).

Reply:
I see, that sounds good. I'll just merge this one, since it's only 37 lines. If you decide that you want to change details, it won't be that painful.
And now that you've written a bunch of code on top of this, that means you've stress-tested it a bit already.